### PR TITLE
 PLATFORM-2484 IPA Testing utilities

### PR DESCRIPTION
**Pull request by zeryx**
working with @adnaanm to build this out to simplify the testing/validation process.

### CODE REVIEW

File: tools/ipa_packageset_builder.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        packages=packages,
+        base_image=base_image,
+        langpacks_version='',
+        langserver_image=LANGSERVER_IMAGE)
+    save_generated_template(generated_template, output_file_path)
+
+    print("completed template construction, file available at {}".format(output_file_path))
+
+
+"""Idk if you wanna use argparse but it's p easy to use."""
```
Comment (pmcq) [inline tools/ipa_packageset_builder.py]: yup would always want to use argparse

File: tools/ipa_packageset_builder.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Creates a packageset dockerfile, by combining package templates together.\n'
+                                                 'Make sure to run this from the root directory.')
+    parser.add_argument('-b', '--base-image', dest='base_image', type=str, required=True)
+    parser.add_argument('-o', '--output-filename', dest='output_path', required=True)
+    parser.add_argument('-p', '--package', action='append',  dest='packages', required=True)
+    args = parser.parse_args()
+    print(args.base_image)
```
Comment (pmcq) [inline tools/ipa_packageset_builder.py]: nit: are these prints really necessary

File: tools/ipa_packageset_builder.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
+from jinja2 import Template
```
Comment (pmcq) [inline tools/ipa_packageset_builder.py]: can you add a `#!/usr/bin/env python3` to the top here

File: tools/ipa_packageset_builder.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        return filepath
+    else:
+        return None
+
+def build(base_image, package_dirs, output_file_path):
+
+    raw_template = get_template(RUNNER_PATH)
+    packages = []
+    for dir in package_dirs:
+        dockerfile_path = path.join('',  *[DIR_PATH_TO_PACKAGES, dir, "Dockerfile"])
```
Comment (pmcq) [inline tools/ipa_packageset_builder.py]: nit: can't you just do `path.join('', DIR_PATH_TO_PACKAGSE, dir, "Dockerfile")`?

File: tools/ipa_packageset_builder.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+RUNNER_PATH = path.join(DIR_PATH_TO_TEMPATES, RUNNER_NAME)
+
+class Package:
+    def __init__(self, package_name, install_script, dockerfile_path, base_image):
+        self.script = install_script
+        if dockerfile_path:
+            self.dockerfile = get_dockerfile_as_string(dockerfile_path)
+        else:
+            raise Exception('dockerfile path not available for package {}'.format(package_name))
+        if base_image:
```
Comment (pmcq) [inline tools/ipa_packageset_builder.py]: i don't think packages need the `base_image`?

File: tools/ipa_packageset_builder.py (hunk 2 above)
Comment (zeryx) [reply tools/ipa_packageset_builder.py]: Nah I kept them for diagnostics, removing em

File: tools/ipa_packageset_builder.py (hunk 4 above)
Comment (zeryx) [reply tools/ipa_packageset_builder.py]: Does that work? I didn't think you could do that with path.join, I'll give er a shot

File: tools/ipa_packageset_builder.py (hunk 5 above)
Comment (zeryx) [reply tools/ipa_packageset_builder.py]: the template doesn't have logic to remove the `FROM {{base_image}}` like an if check, so I assumed it was mandatory.

File: tools/ipa_packageset_builder.py (hunk 1 above)
Comment (zeryx) [reply tools/ipa_packageset_builder.py]: sweet, will continue to use then

File: tools/ipa_packageset_builder.py (hunk 5 above)
Comment (pmcq) [reply tools/ipa_packageset_builder.py]: it needs to be in the context but it's not a property on each package that is referenced (it seems to properly be in the context down below)

**Comment by zeryx** [comment]
Can I merge this without too much pain? @pmcq @neildf

**Review by pmcq — APPROVED** [review]

**Pull request merged**